src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -156,6 +156,32 @@ public IActionResult DrainStatus([FromServices] IScriptHostManager scriptHostMan
             }
         }
 
+        [HttpPost]
+        [Route("admin/host/resume")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> Resume([FromServices] IScriptHostManager scriptHostManager)
+        {
+            _logger.LogDebug($"Received request to resume a draining host - host status: {scriptHostManager?.State.ToString()}");
+
```

Comment:
Because you're accessing State multiple times before the restart and it can change between these accesses, to avoid confusion I think you should capture it once in a local and reuse that state ` var currentState = scriptHostManager.State;`. You don't need the null checks on scriptHostManager either - it's an input parameter from DI so won't be null (see other examples in this file).

Reply:
I think this is addressed, let me know what you think!